nautobot/dcim/models/device_components.py

```diff
@@ -368,6 +372,14 @@ def get_power_draw(self):
                         }
                     )
 
+                    numerator += utilization["allocated_draw_total"] or 0
+                    denominator += utilization["maximum_draw_total"] or 0
```

Comment:
Looking more closely at this, I think these two lines, and the resetting of numerator and denominator back to 0 at lines 356-357, are unnecessary. Lines 342-346 already are taking the sum of allocated/maximum across all grandchild PowerPorts that are connected to the child PowerOutlets dependent on this base PowerPort, independent of which feed leg they're attached to. This inner `for` loop is just calculating the separate per-feed-leg allocated/maximum values - if all child PowerOutlets for this PowerPort are associated with a specific feed leg, this will result in the exact same value as calculated above, and if some child PowerOutlets are themselves three-phase (and hence not associated to a single feed leg) then they will be missed in this recount and the overall total will be incorrect.

Reply:
That makes sense with the look at it. I thought that there may be a case like this and it didn't quite connect the dots though originally.